src/modules/dashboard/components/HookedTokenIcon/HookedTokenIcon.tsx

```diff
@@ -24,76 +26,67 @@ interface Props {
-  fileName &&
-  fileName.substring(fileName.length - 3, fileName.length) === 'svg';
+const TOKEN_LOGOS_REPO_URL =
+  'https://raw.githubusercontent.com/trustwallet' +
+  '/assets/master/blockchains/ethereum/';
 
-const loadTokenImages = async (logo): Promise<ImageType> =>
-  import(
-    /* webpackMode: "eager" */ `../../../../../node_modules/eth-contract-metadata/images/${logo}`
-  );
+const loadTokenImages = async (address: Address): Promise<Response> => {
+  let tokenImageUrl = `${TOKEN_LOGOS_REPO_URL}${address}/logo.png`;
+  if (address === AddressZero) {
+    tokenImageUrl = `${TOKEN_LOGOS_REPO_URL}info/logo.png`;
+  }
+  return fetch(tokenImageUrl);
+};
 
 const HookedTokenIcon = ({
   name,
   token: { iconHash, address },
+  iconName,
   ...props
 }: Props) => {
   const [tokenImage, setTokenImage] = useState<string | undefined>();
-  const [tokenSVG, setTokenSVG] = useState<ImageType['default'] | undefined>();
   const { data: ipfsIcon } = useDataFetcher(
     ipfsDataFetcher,
     [iconHash as string], // Technically a bug, shouldn't need type override
     [iconHash],
   );
 
-  /* Here we will pick the right solution to show
-   *  the token icons depending from if they are png or svg or a blockie
-   *  case 1: if it's a .png we can just pass through the tokenImage
-   *  case 2: it's an svg and we just import it manually
-   *  case 3: there's no data about the token icon and we show a blockie
-   *  case 4: if there's an ipfsHash show the concerning image
-   */
-
   useEffect(() => {
-    const checkAndLoadImages = async () => {
-      if (ipfsIcon || tokenImage || tokenSVG) {
-        return;
-      }
-
-      const metaData = mapping[address];
-
-      if (metaData) {
-        const response = await loadTokenImages(metaData.logo);
-        if (response) {
-          if (checkSVG(metaData.logo)) {
-            setTokenSVG(response.default);
-          } else {
-            setTokenImage(response.default as string);
-          }
+    const loadTokenLogo = async () => {
+      if (address && !iconName) {
+        const response = await loadTokenImages(address);
+        if (!response.ok) {
+          return;
         }
-      }
-      if (address === AddressZero) {
-        const response = await import(
-          /* webpackMode: "eager" */ `../../../../img/tokens/ether.svg`
-        );
-        setTokenSVG(response.default);
+        setTokenImage(response.url);
       }
     };
-    checkAndLoadImages();
-  }, [ipfsIcon, tokenImage, setTokenImage, address, tokenSVG]);
+    loadTokenLogo();
+  }, [address, iconName]);
 
   return (
-    <Avatar
-      avatarURL={tokenImage || ipfsIcon || undefined}
-      placeholderIcon="circle-close"
-      seed={address}
-      title={name || address}
-      {...props}
-    >
-      {tokenSVG}
-    </Avatar>
+    <>
+      {iconName ? (
+        <Icon
+          appearance={{ size: 'medium' }}
+          name={iconName}
+          title={iconName}
+        />
```

Comment:
This overwrites the size of the icons, and it will always display the icon as `medium`, even if we don't want it to:

![Screenshot from 2020-12-10 21-40-54](https://user-images.githubusercontent.com/1193222/101821462-a7865600-3b30-11eb-9f7e-3e4122998eaf.png)

The component already receives a `size` prop, just that you're not passing it down. This should be, as the `Avatar` just passing down all the remaining `props` that the component receives.

Please change this to:
```suggestion
        <Icon name={iconName} title={name || address} {...props} />
```

This way, it will look proper in all instances:

![Screenshot from 2020-12-10 21-41-54](https://user-images.githubusercontent.com/1193222/101821663-ecaa8800-3b30-11eb-94a7-c6384056a3f6.png)